src/group/mls_group/mod.rs

```diff
@@ -376,6 +376,28 @@ impl MlsGroup {
     pub fn group_id(&self) -> &GroupId {
         &self.group_context.group_id
     }
+
+    /// Get the groups extensions.
+    /// Right now this is limited to the ratchet tree extension which is built
+    /// on the fly when calling this function.
+    pub fn extensions(&self) -> Vec<Box<dyn Extension>> {
+        let extensions: Vec<Box<dyn Extension>> = if self.add_ratchet_tree_extension {
```

Comment:
If we have `self.add_ratchet_tree_extension`, shouldn't the extension already be present in somewhere in the group state? I mean as opposed to having to create it ad-hoc. After all, it would also have to be included in the `GroupContext`, which as far as I can see it isn't right now.

Reply:
Yes, it's not supported yet because of #32. We would have to refactor that part obviously, so that we don't generate the extension more often than once per epoch.